packages/plugin-core/src/commands/GotoNote.ts

```diff
@@ -85,9 +88,97 @@ export class GotoNoteCommand extends BasicCommand<CommandOpts, CommandOutput> {
     };
   }
 
+  private getQs(opts: CommandOpts, link: FoundLinkSelection): CommandOpts {
+    if (link.value) {
+      // Reference to another file
+      opts.qs = link.value;
+    } else {
+      // Same file block reference, implicitly current file
+      const note = WSUtils.getActiveNote();
+      if (note) {
+        // Same file link within note
+        opts.qs = note.fname;
+        opts.vault = note.vault;
+      } else {
+        const { wsRoot, vaults } = getDWorkspace().engine;
+        // Same file link within non-note file
+        opts.qs = path.relative(
+          wsRoot,
+          VSCodeUtils.getActiveTextEditorOrThrow().document.fileName
+        );
+        opts.vault = VaultUtils.getVaultByNotePath({
```

Comment:
will this throw an error if `opts.qs` isn't a note?

Reply:
No, it works correctly. It's just computing it based on the file name. I'll go ahead and rename the function since it's serving both purposes now.